src/java.base/share/classes/java/math/BigInteger.java

```diff
@@ -3995,7 +3995,7 @@ final int compareMagnitude(BigInteger val) {
             return 1;
         int i = ArraysSupport.mismatch(m1, m2, len1);
         if (i != -1)
-            return ((m1[i] & LONG_MASK) < (m2[i] & LONG_MASK)) ? -1 : 1;
+            return Integer.compareUnsigned(m1[i], m2[i]) < 0 ? -1 : 1;
```

Comment:
Just an observation.  The (Java and intrinsic) implementation of Integer.compareUnsigned already returns -1, 0, 1.
Returning `Integer.compareUnsigned(m1[i], m2[i])` would yield the same result without the tertiary expression.

Reply:
Yes, that's what was proposed [here](https://github.com/openjdk/jdk/pull/14630#discussion_r1242245875) some time ago.

But the spec of `compareUnsigned()` does _not_ guarantee a -1, 0, 1 result, so there's a (minimal) risk when returning its value directly. (For some reason, `BigInteger` specifies a -1, 0, 1 outcome.)

Reply:
In expressing intent, perhaps `Integer.signum(Integer.compareUnsigned(m1[i], [m2[i]))`.
Though it may all be for nothing if C2 can optimize it away.